src/domains/arbitrum/ArbitrumDomainHost.sol

```diff
@@ -0,0 +1,309 @@
+        );
+    }
+    function deposit(
+        address to,
+        uint256 amount,
+        uint256 maxSubmissionCost,
+        uint256 gasPriceBid
+    ) external payable {
+        deposit(
+            to,
+            amount,
+            maxSubmissionCost,
+            glDeposit,
+            gasPriceBid
+        );
+    }
+
+    function lift(
+        uint256 wad,
+        uint256 maxSubmissionCost,
+        uint256 maxGas,
+        uint256 gasPriceBid
+    ) public payable {
+        (uint256 _rid, uint256 _wad) = _lift(wad);
+        inbox.createRetryableTicket{value: msg.value}(
+            guest,
+            0, // we always assume that l2CallValue = 0
+            maxSubmissionCost,
+            msg.sender,
+            msg.sender,
+            maxGas,
+            gasPriceBid,
+            abi.encodeWithSelector(DomainGuestLike.lift.selector, _rid, _wad)
+        );
+    }
+    function lift(
+        uint256 wad,
+        uint256 maxSubmissionCost,
+        uint256 gasPriceBid
+    ) external payable {
+        lift(
+            wad,
+            maxSubmissionCost,
+            glLift,
+            gasPriceBid
+        );
+    }
+
+    function rectify(
+        uint256 maxSubmissionCost,
+        uint256 maxGas,
+        uint256 gasPriceBid
+    ) public payable {
+        (uint256 _rid, uint256 _wad) = _rectify();
+        inbox.createRetryableTicket{value: msg.value}(
+            guest,
+            0, // we always assume that l2CallValue = 0
+            maxSubmissionCost,
+            msg.sender,
+            msg.sender,
+            maxGas,
+            gasPriceBid,
+            abi.encodeWithSelector(DomainGuestLike.rectify.selector, _rid, _wad)
+        );
+    }
+    function rectify(
+        uint256 maxSubmissionCost,
+        uint256 gasPriceBid
+    ) external payable {
+        rectify(
+            maxSubmissionCost,
+            glLift,
+            gasPriceBid
+        );
+    }
+
+    function cage(
+        uint256 maxSubmissionCost,
+        uint256 maxGas,
+        uint256 gasPriceBid
+    ) public payable {
+        (uint256 _rid) = _cage();
+        inbox.createRetryableTicket{value: msg.value}(
+            guest,
+            0, // we always assume that l2CallValue = 0
+            maxSubmissionCost,
+            msg.sender,
+            msg.sender,
+            maxGas,
+            gasPriceBid,
+            abi.encodeWithSelector(DomainGuestLike.cage.selector, _rid)
+        );
+    }
+    function cage(
+        uint256 maxSubmissionCost,
+        uint256 gasPriceBid
+    ) external payable {
+        cage(
+            maxSubmissionCost,
+            glCage,
+            gasPriceBid
+        );
+    }
+
+    function exit(
+        address usr,
+        uint256 wad,
+        uint256 maxSubmissionCost,
+        uint256 maxGas,
+        uint256 gasPriceBid
+    ) public payable {
+        (address _usr, uint256 _wad) = _exit(usr, wad);
```

Comment:
Returning output variables that are identical to the input variables feels a bit wasteful here, though I guess if the intend is to make it more straightforward for third-parties to implement a bridge for a new domain, then I'm fine with it. That being said, I'd prefer if we could at least ignore these return values in our own Optimism/Arbitrum bridge implementations in favour of using the input variables directly (so `usr` instead of `_usr`, `wad` instead of `_wad`, and similar substitutions for other methods).

Reply:
It is also confusing because it creates false impression that internal function might change the values.

Reply:
I feel returning only the unknown values is probably the most correct.